# Model Evaluation

In [9]:
# Import pandas for table
import pandas as pd

# Create comparison table using actual model results
comparison = pd.DataFrame({
    "Model": ["Logistic Regression", "Random Forest"],
    "Accuracy": [0.74465, 0.73620],
    "Precision (weighted)": [0.741706, 0.734517],
    "Recall (weighted)": [0.74465, 0.73620],
    "F1 Score (weighted)": [0.740253, 0.729080]
})

# Display table
comparison

,Model,Accuracy,Precision (weighted),Recall (weighted),F1 Score (weighted)
0,Logistic Regression,0.74465,0.741706,0.74465,0.740253
1,Random Forest,0.73620,0.734517,0.73620,0.729080


# 📊 === Logistic Regression Evaluation ===

### Overall Performance

- Accuracy: 0.7447  
- Precision (weighted): 0.7417  
- Recall (weighted): 0.7447  
- F1 Score (weighted): 0.7403  

Logistic Regression performs reasonably well on this dataset, with about **74% accuracy**, meaning it correctly predicts most startup outcomes.

---

### Confusion Matrix

| Actual \ Predicted | Acquisition | Failure | IPO |
|-------------------|------------|---------|-----|
| Acquisition       | 5572       | 2839    | 56  |
| Failure           | 1910       | 9211    | 1   |
| IPO               | 300        | 1       | 110 |

---

### Class-wise Performance

#### ✔ Best Performing Class: Failure

- Precision: 0.76  
- Recall: 0.83  
- F1-score: 0.79  

The model performs best at predicting **startup failures**, with strong recall and precision.

---

#### ⚠ Weakest Performing Class: IPO

- Precision: 0.66  
- Recall: 0.27  
- F1-score: 0.38  

The model struggles with IPO prediction due to **class imbalance**, meaning it misses many IPO cases.

---

### Conclusion (Logistic Regression)

- Moderate overall performance (~74%)
- Best at predicting Failure
- Weak at predicting IPO
- Class imbalance is the main limitation

# 🌲 === Random Forest Evaluation ===

### Overall Performance

- Accuracy: 0.7362  
- Precision (weighted): 0.7345  
- Recall (weighted): 0.7362  
- F1 Score (weighted): 0.7291  

Both models perform similarly, with Logistic Regression slightly better overall.

---

### Confusion Matrix

| Actual \ Predicted | Acquisition | Failure | IPO |
|-------------------|------------|---------|-----|
| Acquisition       | 5576       | 2879    | 12  |
| Failure           | 2010       | 9112    | 0   |
| IPO               | 374        | 1       | 36  |

---

### Class-wise Performance

#### ✔ Best Performing Class: Failure

- Precision: 0.76  
- Recall: 0.82  
- F1-score: 0.79  

Failure is the easiest class for both models to predict.

---

#### ⚠ Weakest Performing Class: IPO

- Precision: 0.75  
- Recall: 0.09  
- F1-score: 0.16  

Random Forest performs very poorly on IPO prediction due to strong class imbalance.

---

### Conclusion (Random Forest)

- Similar accuracy to Logistic Regression (~73–74%)
- Both models perform almost the same overall
- Random Forest has much lower IPO recall
- No clear improvement over Logistic Regression

## Feature Importance (Bar Chart) Evaluation

### Top Features Influencing Startup Outcome Prediction

| Feature | Importance |
|----------|------------|
| revenue_million | 0.2684 |
| product_traction_users | 0.2154 |
| founder_experience_years | 0.0857 |
| team_size | 0.0847 |
| market_size_billion | 0.0797 |
| burn_rate_million | 0.0794 |
| funding_rounds | 0.0715 |
| sector | 0.0500 |
| investor_type | 0.0327 |
| founder_background | 0.0326 |

### Key Observations

- **Revenue is the most important feature**, meaning startups with higher revenue strongly influence predictions.
- **Product traction is the second most important feature**, showing user growth matters a lot.
- Features like **founder background and investor type have low importance**, meaning they do not strongly affect predictions.
- Financial and performance-based metrics are more important than categorical features.

---

### Conclusion

The model mainly relies on **revenue** and **user traction**, rather than qualitative founder or investor information. This suggests startup success in this dataset is driven more by measurable business performance than background factors.